### PR TITLE
fix: Fix clipping changes not applying

### PR DESCRIPTION
**Pull request by Youssef1313**
GitHub Issue (If applicable): closes #

## PR Type

What kind of change does this PR introduce?

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/SamplesApp/UITests.Shared/Windows_UI_Xaml/Clipping/UIElement_Clip_Update.xaml.cs [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+namespace UITests.Windows_UI_Xaml.Clipping;
+
+
+[Sample("Clipping", Description = """
+	1. Initially, there should be a button and a 100x100 border.
+	2. Click the button.
+	3. The border should become smaller, 80x80
+	4. Click the button once again
+	5. The border should become smaller, 50x50
+	""", IsManualTest = true)]
```
Comment (Youssef1313) [inline src/SamplesApp/UITests.Shared/Windows_UI_Xaml/Clipping/UIElement_Clip_Update.xaml.cs:16]: This is a manual test due to a bug in RenderTargetBitmap where it shows the border as being clipped when it's not (when the InvalidateArrange is missing). We have no other good way to test this AFAIK.

File: src/Uno.UI/UI/Xaml/FrameworkElement.Layout.crossruntime.cs [hunk 2]
```diff
@@ -735,7 +735,12 @@ private void ArrangeNative(Point offset, bool needsClipToSlot, Rect clippedFrame
 				throw new InvalidOperationException($"{FormatDebugName()}: Invalid frame size {newRect}. No dimension should be NaN or negative value.");
 			}
 
-			var clipRect = Clip?.Rect ?? (needsClipToSlot ? clippedFrame : default(Rect?));
+			var clip = Clip;
+			var clipRect = clip?.Rect ?? (needsClipToSlot ? clippedFrame : default(Rect?));
+			if (clipRect.HasValue && clip?.Transform is { } transform)
+			{
+				clipRect = transform.TransformBounds(clipRect.Value);
+			}
```
Comment (dr1rrb) [inline src/Uno.UI/UI/Xaml/FrameworkElement.Layout.crossruntime.cs:743]: @carldebilly

**Review by dr1rrb — APPROVED** [review]

File: src/Uno.UI/UI/Xaml/FrameworkElement.Layout.crossruntime.cs (hunk 2 above)
Comment (carldebilly) [reply src/Uno.UI/UI/Xaml/FrameworkElement.Layout.crossruntime.cs:743]: Seems legit.

**Review by carldebilly — APPROVED** [review]

**Pull request merged**